File: Jan/Week 2/LeetCode/problem_7.cpp
//  Day 14 Problem No 28

// 345. Reverse Vowels of a String => https://leetcode.com/problems/reverse-vowels-of-a-string/description/?envType=study-plan-v2&envId=leetcode-75

class Solution {
public:
    string reverseVowels(string s) {
        string v = "aeiouAEIOU";
        int st = 0;
        int end = s.length() -1;
        set<char> v_s(v.begin(),v.end());
        while(st < end){
            while(st < end && v_s.find(s[st]) == v_s.end()){
                st++;
            }
            while(st < end && v_s.find(s[end]) == v_s.end()){
                end--;
            }
            char tmp = s[st];
            s[st] = s[end];
            s[end] = tmp;

            st++;
            end--;
        }
        return s;
    }
};

File: Jan/Week 3/LeetCode/problem_7.py
# Day 21 Problem No 45

# 2215. Find the Difference of Two Arrays => https://leetcode.com/problems/find-the-difference-of-two-arrays/description/?envType=study-plan-v2&envId=leetcode-75

class Solution:
    def findDifference(self, nums1: List[int], nums2: List[int]) -> List[List[int]]:
        l1 = set(nums1)
        l2 = set(nums2)
        
        a1 = list(l1 -l2)
        a2 = list(l2 -l1)

        return [list(a1), list(a2)]

# C++ Solution

# class Solution {
# public:
#     vector<vector<int>> findDifference(vector<int>& nums1, vector<int>& nums2) {
#         set<int> set1(nums1.begin(),nums1.end());
#         set<int> set2(nums2.begin(),nums2.end());

#         vector<int> d1;
#         vector<int> d2;

#         for(int num:set1){
#             if(set2.find(num) == set2.end()){
#                 d1.push_back(num);
#             }
#         }

#         for(int num:set2){
#             if(set1.find(num) == set1.end()){
#                 d2.push_back(num);
#             }
#         }
#         return {d1,d2};
#     }
# };